src/test/regress/specs/isolation_multi_shard_modify_vs_all.spec

```diff
@@ -0,0 +1,125 @@
+	SELECT create_distributed_table('events_test_table', 'user_id');
+	INSERT INTO events_test_table VALUES
+	(1, 5, 7, 7),
+	(3, 11, 78, 18),
+	(5, 22, 9, 25),
+	(7, 41, 10, 23),
+	(9, 34, 11, 21),
+	(1, 20, 12, 25),
+	(3, 26, 13, 18),
+	(5, 17, 14, 4);
+}
+
+teardown
+{
+	DROP TABLE users_test_table;
+	DROP TABLE events_test_table;
+	SELECT citus.restore_isolation_tester_func();
+}
+
+session "s1"
+
+step "s1-begin"
+{
+    BEGIN;
+}
+
+step "s1-change_connection_mode_to_sequential"
+{
+    set citus.multi_shard_modify_mode to 'sequential';
+}
+
+step "s1-update_all_value_1"
+{
+	UPDATE users_test_table SET value_1 = 3;
+}
+
+step "s1-update_value_1_of_1_or_3"
+{
+	UPDATE users_test_table SET value_1 = 5 WHERE user_id = 1 or user_id = 3;
+}
+
+step "s1-commit"
+{
+    COMMIT;
+}
+
+session "s2"
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-change_connection_mode_to_sequential"
+{
+    set citus.multi_shard_modify_mode to 'sequential';
+}
+
+step "s2-select" 
+{
+	SELECT * FROM users_test_table ORDER BY value_1;
+}
+
+step "s2-insert-to-table"
+{
+	INSERT INTO users_test_table VALUES (1,2,3,4);
+}
+
+step "s2-insert-into-select"
+{
+	INSERT INTO users_test_table SELECT * FROM events_test_table;
+}
+
+step "s2-update_all_value_1"
+{
+	UPDATE users_test_table SET value_1 = 6;
+}
+
+step "s2-update_value_1_of_1_or_3"
+{
+	UPDATE users_test_table SET value_1 = 8 WHERE user_id = 1 or user_id = 3;
+}
+
+step "s2-update_value_1_of_5_or_7"
+{
+	UPDATE users_test_table SET value_1 = 4 WHERE user_id = 5 or user_id = 7;
+}
+
+step "s2-commit"
+{
+	COMMIT;
+}
+
+# test with parallel connections
+permutation "s1-begin" "s1-update_all_value_1" "s2-begin" "s2-select" "s1-commit" "s2-select" "s2-commit"
+permutation "s1-begin" "s1-update_all_value_1" "s2-begin" "s2-update_all_value_1" "s1-commit" "s2-commit"
+permutation "s1-begin" "s1-update_value_1_of_1_or_3" "s2-begin" "s2-update_value_1_of_5_or_7" "s1-commit" "s2-commit" "s2-select" 
+permutation "s1-begin" "s1-update_value_1_of_1_or_3" "s2-begin" "s2-update_value_1_of_1_or_3" "s1-commit" "s2-commit" "s2-select" 
+permutation "s1-begin" "s1-update_all_value_1" "s2-begin" "s2-insert-to-table" "s1-commit" "s2-commit" "s2-select" 
+permutation "s1-begin" "s1-update_all_value_1" "s2-begin" "s2-insert-into-select" "s1-commit" "s2-commit" "s2-select"
+
+# test with parallel connections
+permutation "s1-begin" "s1-change_connection_mode_to_sequential" "s1-update_all_value_1" "s2-begin" "s2-change_connection_mode_to_sequential" "s2-update_all_value_1" "s1-commit" "s2-commit" "s2-select"
+permutation "s1-begin" "s1-change_connection_mode_to_sequential" "s1-update_value_1_of_1_or_3" "s2-begin" "s2-change_connection_mode_to_sequential" "s2-update_value_1_of_1_or_3" "s1-commit" "s2-commit" "s2-select"
+permutation "s1-begin" "s1-change_connection_mode_to_sequential" "s1-update_value_1_of_1_or_3" "s2-begin" "s2-change_connection_mode_to_sequential" "s2-update_value_1_of_5_or_7" "s1-commit" "s2-commit" "s2-select"
```

Comment:
It's useful to add a comment briefly describing the test, and what your expectation is for the outcome of the test.

e.g. "sequential multi-shard UPDATEs should not block each other"